lib/routes/weibo/user.js

```diff
@@ -39,10 +39,22 @@ module.exports = async (ctx) => {
         },
     });
 
+    let earliestDate;
+
     const resultItem = await Promise.all(
         response.data.data.cards
-            .filter((item) => item.mblog)
+            .reverse() // reverse first to ensure the pinned card will be processed lastly
```

Comment:
What happened if a Weibo user only has one tweet and it is pinned?

Reply:
That's safe. Please check [L77](https://github.com/DIYgod/RSSHub/blob/d6c3e34d17f56bdaf9e25fabb8a5365d97bea7e0/lib/routes/weibo/user.js#L77). If there is only one weibo and it's pinned, `earliestDate` remains `undefined`, thus the weibo will not be dropped.